### PR TITLE
Update release workflow to work with new branch protections

### PR DESCRIPTION
**Pull request by austinvalle**
- Contributes to: https://github.com/hashicorp/terraform-providers-devex-internal/issues/136

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -70,28 +69,40 @@ jobs:
...
+          git push "https://${{ env.CI_COMMIT_AUTHOR }}:${{ secrets.TF_DEVEX_COMMIT_GITHUB_TOKEN }}@github.com/${{ github.repository }}.git" "${{ inputs.versionNumber }}"
+          git push "https://${{ env.CI_COMMIT_AUTHOR }}:${{ secrets.TF_DEVEX_COMMIT_GITHUB_TOKEN }}@github.com/${{ github.repository }}.git" -f "${{ needs.major-version.outputs.version }}"
 
   release:
     needs: [ changelog-version, changelog, release-tag ]
-    runs-on: "ubuntu-latest"
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # Needed to create GitHub release
+      issues: write # Needed to close associated milestone
```
Comment (austinvalle) [inline .github/workflows/release.yml:94]: Do we actually need this permission? As far as I can tell, we don't close the milestone in this github workflow and GH cli doesn't support it natively: https://github.com/cli/cli/issues/1200

Comment (bflad) [reply .github/workflows/release.yml:94]: We can skip this here for now -- if we do want to manage milestones in these NodeJS projects on release, we can figure out the appropriate solution then.

**Review by bflad — APPROVED** [review]
Looks good to me 🚀

File: .github/workflows/release.yml (hunk 1 above)
Comment (austinvalle) [reply .github/workflows/release.yml]: Removed ⚡

**Pull request merged**